app/src/main/java/org/luxoft/sdl_core/SdlLauncherService.java

```diff
@@ -135,41 +138,79 @@ private void showToastMessage(String message) {
+        intent.putExtra(IS_SDL_SERVICE_THREAD_ALIVE, isThreadAlive);
+
+        PendingIntent contentIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (channel_id == null) {
+                Log.w(TAG, "Cannot create notification. Create and save channel_id firstly");
+                return null;
+            }
+            return new NotificationCompat.Builder(this, channel_id)
+                    .setContentTitle(SDL_CONTENT_TITLE)
+                    .setContentText(text)
+                    .setOnlyAlertOnce(true) // so when data is updated don't make sound and alert in android 8.0+
+                    .setOngoing(true)
+                    .setSmallIcon(R.drawable.ic_launcher_foreground)
+                    .setContentIntent(contentIntent)
+                    .build();
+        } else {
+            return new NotificationCompat.Builder(this)
+                    .setContentTitle(SDL_CONTENT_TITLE)
+                    .setContentText(text)
+                    .setOngoing(true)
+                    .setSmallIcon(R.drawable.ic_launcher_foreground)
+                    .setContentIntent(contentIntent)
+                    .build();
+        }
     }
 
-    @SuppressLint("NewApi")
-    public void enterForeground() {
+    private void enterForeground() {
         NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
-        if (notificationManager != null) {
+        if (notificationManager == null) {
+            Log.w(TAG, "NotificationManager are not available. Notification skipped.");
+            return;
+        }
+        final String text = "SDL core is starting...";
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
             if (channel_id == null) {
                 NotificationChannel channel = new NotificationChannel(APP_ID, SERVICE_NAME, NotificationManager.IMPORTANCE_DEFAULT);
                 notificationManager.createNotificationChannel(channel);
                 channel_id = channel.getId();
             }
 
-            startForeground(FOREGROUND_SERVICE_ID, getServiceNotification("SDL core is starting..."));
+            startForeground(FOREGROUND_SERVICE_ID, getServiceNotification(text));
+        } else {
+            updateServiceNotification(text);
+        }
+    }
+
+    private void exitForeground() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            stopForeground(true);
+        } else {
+            NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+            if (notificationManager == null) {
+                Log.w(TAG, "NotificationManager are not available. Notification cannot be canceled.");
+                return;
+            }
+            notificationManager.cancel(FOREGROUND_SERVICE_ID);
```

Comment:
@YaroslavLutsenko don't we need to cancel the notification for newer Android versions as well?

Reply:
@AKalinich-Luxoft `stopForeground(true);` will remove notification for newer Android versions.